x/ibc/03-connection/types/version.go

```diff
@@ -1,39 +1,120 @@
 package types
 
 import (
+	"fmt"
+	"strings"
+
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
 var (
-	// DefaultIBCVersion represents the latest supported version of IBC
-	DefaultIBCVersion = "1.0.0"
+	// DefaultIBCVersion represents the latest supported version of IBC.
+	// The current version supports only ORDERED and UNORDERED channels.
+	DefaultIBCVersion = CreateVersionString("1", []string{"ORDERED channel", "UNORDERED channel"})
 )
 
-// GetCompatibleVersions returns a descending ordered set of compatible IBC versions
-// for the caller chain's connection end. The latest supported version should be
-// first element and the set should decsend to the oldest supported version.
+// CreateVersionString constructs a valid connection version given a
+// version number and feature set. The format is written as:
+// [version-number]-[feature_0,feature_1,feature_2...]
```

Comment:
I think this should be: `([version_identifier], [feature_set])`

so I guess example versions could be:
`(1, [Ordered channels])`
`(1, [Ordered channels, Unordered channels])`
`(2, [])`

Reply:
I'll add some more documentation as well that defining two version identifiers with different feature sets in your supported versions can result in undefined behavior and therefore should be avoided.